ignition_ros2_control/src/ignition_system.cpp

```diff
@@ -0,0 +1,451 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ignition_ros2_control/ignition_system.hpp"
+
+#include <ignition/msgs/imu.pb.h>
+
+#include <ignition/gazebo/components/AngularVelocity.hh>
+#include <ignition/gazebo/components/Imu.hh>
+#include <ignition/gazebo/components/JointForce.hh>
+#include <ignition/gazebo/components/JointForceCmd.hh>
+#include <ignition/gazebo/components/JointPosition.hh>
+#include <ignition/gazebo/components/JointPositionReset.hh>
+#include <ignition/gazebo/components/JointVelocity.hh>
+#include <ignition/gazebo/components/JointVelocityCmd.hh>
+#include <ignition/gazebo/components/LinearAcceleration.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/gazebo/components/Pose.hh>
+#include <ignition/gazebo/components/Sensor.hh>
+
+#include <ignition/transport/Node.hh>
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+struct jointData
+{
+  /// \brief vector with the joint's names.
+  std::string name;
+
+  /// \brief vector with the current joint position
+  double joint_position;
+
+  /// \brief vector with the current joint velocity
+  double joint_velocity;
+
+  /// \brief vector with the current joint effort
+  double joint_effort;
+
+  /// \brief vector with the current cmd joint position
+  double joint_position_cmd;
+
+  /// \brief vector with the current cmd joint velocity
+  double joint_velocity_cmd;
+
+  /// \brief vector with the current cmd joint effort
+  double joint_effort_cmd;
+
+  /// \brief handles to the joints from within Gazebo
+  ignition::gazebo::Entity sim_joint;
+
+  /// \brief vector with the control method defined in the URDF for each joint.
+  ignition_ros2_control::IgnitionSystemInterface::ControlMethod joint_control_method;
+};
+
+class ImuData
```

Comment:
Thanks for adding IMU. Would it make sense to add this as a separate plugin of ros2_control hardware interface?

Reply:
Openned an issue https://github.com/ignitionrobotics/ign_ros2_control/issues/13